presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -446,12 +471,36 @@ public static long toPrestoTimestamp(LocalDateTime localDateTime)
     @Deprecated
     public static LocalDateTime fromPrestoLegacyTimestamp(long value, ZoneId sessionZone)
     {
-        return Instant.ofEpochMilli(value).atZone(sessionZone).toLocalDateTime();
+        return fromPrestoLegacyTimestamp(createTimestampType(3), value, sessionZone);
+    }
+
+    /**
+     * @deprecated applicable in legacy timestamp semantics only
+     */
+    @Deprecated
+    public static LocalDateTime fromPrestoLegacyTimestamp(TimestampType timestampType, long value, ZoneId sessionZone)
+    {
+        return getInstantFromTimestamp(timestampType, value).atZone(sessionZone).toLocalDateTime();
     }
 
     public static LocalDateTime fromPrestoTimestamp(long value)
     {
-        return Instant.ofEpochMilli(value).atZone(UTC).toLocalDateTime();
+        return fromPrestoTimestamp(createTimestampType(3), value);
```

Comment:
This is called for every row, so use the TIMESTAMP const, or require that `createTimestampType(3)` be provided by the caller. (or mark the method deprecated and we remove it soon)